llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -2352,6 +2359,78 @@ Instruction *ModuleAddressSanitizer::CreateAsanModuleDtor(Module &M) {
   return ReturnInst::Create(*C, AsanDtorBB);
 }
 
+void ModuleAddressSanitizer::instrumentSyclDeviceGlobals(IRBuilder<> &IRB,
+                                                         Module &M) {
+  auto &DL = M.getDataLayout();
+  SmallVector<GlobalVariable *, 8> GlobalsToRemove;
+  SmallVector<GlobalVariable *, 8> NewDeviceGlobals;
+  SmallVector<Constant *, 8> DeviceGlobalMetadata;
+
+  // Device global meta data is described by a structure
+  //  size_t device_global_size
+  //  size_t device_global_size_with_red_zone
+  //  size_t beg
```

Comment:
How about using a more descriptive variable name for "beg"?

Reply:
done